analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ObjectsShouldNotBeDisposedMoreThanOnceBase.cs

```diff
@@ -18,10 +18,31 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
 
 public abstract class ObjectsShouldNotBeDisposedMoreThanOnceBase : SymbolicRuleCheck
 {
     protected const string DiagnosticId = "S3966";
-    protected const string MessageFormat = "Refactor this code to make sure '{0}' is disposed only once.";
+    protected const string MessageFormat = "Resource '{0}' has already been disposed explicitly or implicitly through a using statement. Please remove the redundant disposal.";
+
+    private static readonly string[] DisposeMethods = { "Dispose", "DisposeAsync"};
+
+    protected override ProgramState PreProcessSimple(SymbolicContext context)
+    {
+        var state = context.State;
+        if (context.Operation.Instance.AsInvocation() is { } invocation && Array.Exists(DisposeMethods, x => x.Equals(invocation.TargetMethod.Name)))
```

Comment:
We have array with 2 members. The `Array.Exists` is totally not worth the syntax trouble and the memory footprint of creating the lambda.
Either revert it to simple `.Contains` (there will be no significant impact on performance for array with 2 members), or get rid of the array and use `=="Dispose" || ..=="DisposeAsync")

Reply:
I kept the array because "close" is supposed to be added too there - and it's better to make the array larger rather than the condition.